lesson_12/homework.md

```diff
@@ -0,0 +1,170 @@
+# Homework: Classes
+
+🔥 **Read carefully until the end before you start solving the exercises.** 🔥
+
+## Practice the Basics 💪🏻
+
+### Basic Class
+
+- Create an empty class `HouseForSale`
+- Create two instances.
+- Add `number_of_rooms` and `price` as instance attributes.
+- Create print statements that show the attribute values for the 
+instances.
+
+### Instance Methods
+
+- Create a `Computer` class.
+- Create method:
+  - `turn_on` that prints `Computer has Turned On`
+  - `turn_off` that prints `Computer has Turned Off`
+- Create an instance of the `Computer` class then call each method.
+
+### Constructor with Parameters
+
+- Create a `Dog` class.
+- `Dog` should have a constructor with a `name` parameter.
+- `Dog` should have a method `say_name` that prints the name of the dog.
+
+### Inheritance
+
+Create the classes that would make the following code possible, with the caveat
+that both `Dog` and `Cat` must inherit from `Animal`.
+
+```python
+animal = Animal()
+animal.say_name()   # Prints: I don't have a name yet.
+animal.speak()      # Prints: I can't speak!
+
+dog = Dog('Fido')
+dog.say_name()      # Prints: Fido
+dog.speak()         # Prints: Woof!
+
+cat = Cat('Max')
+cat.say_name()      # Prints: Max
+cat.speak()         # Prints: Meow!
+```
+
+## Exercises 🏆
+
+###  Exercise 1: Books and Authors
+
+Create an empty class called Book. Then create three instances.
+Add the following attributes for each of the instances: title, author, and publication_year.
+Create print statements to display the attributes of each one of the instances.
+
+```python
+# Pre-code:
+class Book:
+   pass
+
+book1 = Book()
+book1.t??? = 'To Kill a Mockingbird'
+book1.a??? = 'Harper Lee'
+book1.p??? = 1960
+
+# Your code here
+```
+
+### Exercise 2. Vehicle and Types of Vehicles
+
+Create a Vehicle class. 
+
+- Its constructor should take the name and type of the vehicle and store them as instance attributes. 
+- This `Vehicle` class should also have a `show_type()` instance method that prints out the 
+message: `"<NAME_OF_VEHICLE> is a <TYPE_OF_VEHICLE>"`
+- Create `Car` and `Bike` classes that inherit from `Vehicle`.
+- Create instances of `Car` and `Bike` and make them show their types.
+
+### Exercise 3. Spot and correct the mistakes
+
+- You are given a task to create a Car class. 
+- Each car will have attributes for model and year. 
+- Unfortunately, the given code below contains several mistakes. 
+- Your task is to find and correct these mistakes to make the code run successfully.
```

Comment:
It might be a good idea to ask the students to leave a comment next to the correction they made. Maybe we could mention it as part of instructions?
Somehting like:
- Leave a comment next to the corrections you've made explaining what was done (feel free to paraphrase it)

Reply:
Will add the following comment as part of the instructions:

"**Please include a comment in the code explaining the corrections you made and why.**"